internal/model/board.go

```diff
@@ -0,0 +1,134 @@
+package model
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+)
+
+const (
+	IShapedPercentage                = 0.26
+	TShapedPercentage                = 0.36
+	VShapedWithoutTreasurePercentage = 0.20
+	VShapedWithTreasurePercentage    = 0.18
+)
+
+var (
+	randomGenerator = rand.New(rand.NewSource(time.Now().UnixNano()))
+)
+
+// Board represents the game board.
+type Board struct {
+
+	// Tiles are the tile sthat are placed on a board.
+	Tiles [][]*BoardTile
+
+	// RemainingTile is the tile that was not placed on the board.
+	RemainingTile *Tile
```

Comment:
I believe it should be a BoardTile, as it can be rotated. 

My reasoning is that the game state will be modified by each user action - including rotating the remaining tile.